src/Core/MainLoop.js

```diff
@@ -68,7 +68,18 @@ function updateElements(context, geometryLayer, elements) {
         // update attached layers
         for (const attachedLayer of geometryLayer._attachedLayers) {
             if (attachedLayer.ready) {
-                attachedLayer.update(context, attachedLayer, element);
+                const sub = geometryLayer.getObjectToUpdateForAttachedLayers(element);
+
+                if (sub) {
+                    if (__DEBUG__) {
+                        if (!(sub.element.isObject3D)) {
+                            throw new Error(`
+                                Invalid object for attached layer to update.
+                                Must be a THREE.Object and have a THREE.Material`);
+                        }
+                    }
```

Comment:
the above does not depend on attachedLayer and could be moved outside the loop

Reply:
I missed this, thanks.

Reply:
Fixed.